### ISSUE TITLE
Virtualdub frame server

### ISSUE THREAD

**Issue by Gabarito007** [stax76/staxrip#159]
Is there a way to implement Staxrip to receive Virtualdub frame server files?
Some filters, like Deshaker and others, are available only in Vdub.
I could use some lossless codec to make an intermediate file and send it to Staxrip after that, but with frame server it would be far better. No disk space needed neither waste of time.
I use to create an vdr-avi file, but Staxrip will refuse it.

Log:
[aaa.vdr_StaxRip.log.txt](https://github.com/stax76/staxrip/files/977065/aaa.vdr_StaxRip.log.txt)

**Comment by stax76** [comment]
Do you know if there is a AviSynth filter that can open it?

**Comment by Gabarito007** [comment]
I know that TMPGEnc can handle that dummy file.
I already try it with Vegas someday and it worked.
No info about Avisynth source filter.

But it would be WONDERFUL if Staxrip could handle Vdub frame server. 👍

**Comment by Gabarito007** [comment]
I tried just now and [AvsPmod](https://avspmod.github.io/) can open that dummy!!!

**Comment by Gabarito007** [comment]
Making some tests.

Started Virtualdub
Open test.mp4 file
Started Frame Server with dummy file test.vdr.avi
I created an avs file with just one line: Avisource("test.vdr.avi")
Save it as test.avs
Open it with AvsPmod to check if it works -> OK
Started other Virtualdub instance
Open test.avs to check again if it works -> OK
Started Staxrip stable version
Try to open that test.avs using Avisinth+ filter -> Fail...
:-(

Error message:
Failed to load script.
AVIFileSource: couldn't open file 'B:\Meus documentos\Downloads\test.vdr.avi'
(B:\Programas\StaxRipTemp\test_temp\test_Source.avs, line 1)

Why? That avs file works!...

OK, trying to load that test.avs file with Staxrip using Avisource filter, after install python-3.5.1 and VapourSynth-R35 -> Program stop working (System message)

Message:
Assinatura do problema:
  Nome do Evento de Problema:	APPCRASH
  Nome do Aplicativo:	StaxRip.exe
  Versão do Aplicativo:	1.4.0.0
  Carimbo de Data/Hora do Aplicativo:	584f1657
  Nome do Módulo de Falhas:	ucrtbase.DLL
  Versão do Módulo de Falhas:	10.0.10240.16390
  Carimbo de Data/Hora do Módulo de Falhas:	55a5b718
  Código de Exceção:	40000015
  Deslocamento de Exceção:	0000000000065a5f
  Versão do sistema operacional:	6.1.7601.2.1.0.256.1
  Identificação da Localidade:	1046
  Informações Adicionais 1:	eda2
  Informações Adicionais 2:	eda2e24cc7408a1bbee8881a667641e0
  Informações Adicionais 3:	6ace
  Informações Adicionais 4:	6ace54f5e1c0b92d7c254d9e4947eebf

I still hope that there is a solution to load Vdub frame server...

**Comment by Gabarito007** [comment]
Another program can open that test.avs file:
Avidemux, using its avsproxy_gui tool.
It asked to append ConvertToYV2() function, but it could open the video.

**Comment by stax76** [comment]
Are you sure all tools are x64?

**Comment by Gabarito007** [comment]
I'm using Windows Seven 64 bits SP1.
I'm not sure about the tools.
I think that Avidemux 2.5.4, TMPGEnc 2.524 and AvsPmod 2.5.1 are 32 bits.
How to know for sure if x32 or x64?
Virtualdub 1.10.4 is x32 because there are 2 flavors at download page.

**Comment by stax76** [comment]
That's the reason then, to know rather an app is x86 or x64 you can use task manager.

If there are no x64 tools the only possibility is using a virtual file system, both avisynth and vapoursynth support it.

**Comment by Gabarito007** [comment]
I don't understand exactly what you said.
I run Vdub64 and started frame server.
Staxrip still is unable to load dummy vdr-avi file or avs file.
Trying Avisinth+, Vapoursynth and Avisource filters.
What do you mean with virtual file system by avisynth and vapoursynth?

**Comment by stax76** [comment]
Took me a while but finally found the clue in here:

C:\Program Files\VirtualDub\VirtualDub.chm

> Installing and using the 64-bit frameserver and frameclient
> Use of the frameserver and frameclient with 64-bit programs, including the 64-bit version of VirtualDub, requires special procedures.
> 
> The 32-bit version of VirtualDub, and the 32-bit frameserver/frameclient libraries, will work without modification under Windows x64 Edition. However, frameserving with 64-bit programs involved requires the 64-bit versions of the frameserver/frameclient libraries to be installed. Doing this currently requires manual steps:
> 
> Copy vdsvrlnk64.dll and vdremote64.dll to the Windows\System32 folder. 
> Install the frameserver64.reg patch into the Registry.
> Once this is done, the 64-bit programs can interact with the frameserver system. This includes crossing data between 32-bit and 64-bit applications. In addition to frameserving between 64-bit applications, it is also possible to frameserve from 64-bit VirtualDub to a 32-bit application, as well as from 32-bit VirtualDub to a 64-bit application.

copy this:

C:\Program Files\VirtualDub\vdsvrlnk64.dll
C:\Program Files\VirtualDub\vdremote64.dll

to here:

C:\Windows\System32 (Tools > Directories > System)

also use vdr as your frameserver file extension

**Comment by Gabarito007** [comment]
FAN-TAS-TIC!!!

Stax76, my congratulations, felicitations, anything else, for your attention to kindly hear users of your program! Any feedback is very important, but you put it in a higher level very above.

It works like a charm and now I can use all nice Vdub filters and send edited video to Staxrip.

Excellent!!!!!

:-)
:-D

**Comment by stax76** [comment]
I'm glad to help. With the next build you can open vdr files directly.

**Comment by Gabarito007** [comment]
Hi, Stax.

Sorry to reopen this issue...
I had to reinstall whole Windows and some settings were gone.
I use to send videos via FrameServer without problem all this past time.
I'm trying to use FrameServer, but I get this, when picking AviSource:

> Failed to open source. Try another source filter?
AviFileSource: Couldn't open file noname.vdr.avi
noname_source.avi, line 1.

I'm still using version x64 1.5.0.0 stable release.
After system restore, I had to install first
dotNET 4.7 - NDP47-KB3186497-x86-x64-AllOS-ENU
Visual C++ 2017 - vc_redist.x64

Avisynth 2.60 and Avisynth+ r2455 also installed.

What is going wrong this time?
